src/compiler/scanner.ts

```diff
@@ -4012,6 +4016,10 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
     function setSkipJsDocLeadingAsterisks(skip: boolean) {

```

Comment:
why doesn't setSkipJsDocLeadingAsterisks suffice? Looking briefly at the parser code, it looks like it's set correctly. Is the position returned by getNodePos incorrect when it's set? In that case, probably the fix needs to be there instead of createIdentifier.

Edit: that's probably it; getNodePos returns scanner.getTokenFullStart; this PR switches to getTokenStart. But if skipJsDocLeadingAsterisks was set, it seems to me that getTokenFullStart should also skip the leading asterisk -- and return the same position as getTokenStart.

Reply:
@sandersn Thanks for the detailed feedback 👍. I'll look into how we can add these changes to the scanner. If I remember correctly, the end position relies on `getTokenFullStart`. If we adjust `fullStartPos` after `*`, we might run into situations where `*` appears after names. or am I missing something here?

```ts
/**
 * @import {    
 *    Foo       // node pos + end (fullStartPos after *) "Foo * " ?
 * } from "./a"
 */

Reply:
No, your memory is probably better than mine on this front. Still, thanks for looking into it.